dartagnan/src/main/java/com/dat3m/dartagnan/program/event/Tag.java

```diff
@@ -259,5 +236,29 @@ public static final class IMM {
         private IMM() {}
 
         public static final String CASDEPORIGIN = "CASDEPORIGIN";
+
+        public static String extractStoreMo(String cMo) {
+        	switch(cMo) {
+        		case C11.MO_SC:
+        		case C11.MO_ACQUIRE_RELEASE:
+        			return C11.MO_RELEASE;
+        		case C11.MO_ACQUIRE:
+        			return C11.MO_RELAXED;
+        		default:
+        			return cMo;
+        	}
+        }
```

Comment:
Why is this code now in IMM?

Reply:
There are 3 C-like models: C11, RC11 and IMM. The first two understand all memory orders (RC11 does not really understand `CON`), thus it is only IMM which needs to convert SC to something else. We are doing pretty much the same for hardware models. Maybe `extractXMoFromCMo` would be a better name to be consistent with the rest.